uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -175,6 +178,26 @@ define([
       section.dataIndex = _.indexBy(section.data, 'id');
     }
 
+    /**
+     * Initiates the role based on permissions.
+     * @NOTE It skips the permission check if the section is 'absence-tab'
+     * because only admins can access this section.
+     */
```

Comment:
1. Return is not documented.
2. When `$rootScope.section === 'absence-tab'`, the function is not returning a promise. But the function return type should always be of same type.